pkg/graph/resolvers/trb_user_info_dataloader.go

```diff
@@ -0,0 +1,97 @@
+package resolvers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/graph-gophers/dataloader"
+
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+// Loaders wrap your data loaders to inject via middleware
+type Loaders struct {
+	UserInfoLoader *dataloader.Loader
+}
+
+type ctxKey string
+
+const (
+	loadersKey = ctxKey("dataloaders")
+)
+
+// UserInfoFetcher reads Users from a database
+type UserInfoFetcher struct {
+	FetchUserInfos func(context.Context, []string) ([]*models.UserInfo, error)
+}
+
+// BatchUserInfos implements a batch function to populate UserInfo for EUA IDs
+func (u *UserInfoFetcher) BatchUserInfos(
+	ctx context.Context,
+	keys dataloader.Keys,
+) []*dataloader.Result {
+	results := make([]*dataloader.Result, len(keys))
+	euas := make([]string, len(keys))
+	for i, key := range keys {
+		euas[i] = key.String()
+	}
+
+	// Maps EUAs to UserInfo structs
+	euaUserInfoMap := map[string]*models.UserInfo{}
+	userInfos, err := u.FetchUserInfos(ctx, euas)
+	if err != nil {
+		return results
+	}
+
+	for _, userInfo := range userInfos {
+		euaUserInfoMap[userInfo.EuaUserID] = userInfo
+	}
+
+	for i, key := range keys {
+		if userInfo, ok := euaUserInfoMap[key.String()]; ok {
+			results[i] = &dataloader.Result{Data: userInfo, Error: nil}
+		} else {
+			err := fmt.Errorf("No user info found for EUA ID %s", key.String())
+			results[i] = &dataloader.Result{Data: nil, Error: err}
+		}
+	}
+	return results
+}
+
+// NewLoaders instantiates data loaders for the middleware
+func NewLoaders(fetchUserInfos func(context.Context, []string) ([]*models.UserInfo, error)) *Loaders {
+	userInfoFetcher := UserInfoFetcher{
+		FetchUserInfos: fetchUserInfos,
+	}
+	loaders := &Loaders{
+		UserInfoLoader: dataloader.NewBatchedLoader(userInfoFetcher.BatchUserInfos),
+	}
+	return loaders
+}
+
+// Middleware injects data loaders into the context
+func Middleware(loaders *Loaders, next http.Handler) http.Handler {
```

Comment:
like above comments, it would be good to remove the duplicate code here that is in the middleware package.

Reply:
I am not quite sure how, but you have reviewed the wrong commit - there is no such duplicate code in the most recent commit. This file does not actually exist in the most recent commit - it's called user_info_loader.go. Weird, because I have not made any changes since you made your review.